modules/settings/settings.php

```diff
@@ -461,7 +461,7 @@ public function helper_settings_validate_and_save()
             foreach ($modules as $moduleSlug) {
                 $module_name = sanitize_key(Util::sanitizeModuleName($moduleSlug));
 
-                $new_options = (isset($_POST[$legacyPlugin->$module_name->module->options_group_name])) ? $_POST[$legacyPlugin->$module_name->module->options_group_name] : [];
+                $new_options = (isset($_POST[$legacyPlugin->$module_name->module->options_group_name])) ? sanitize_text_field($_POST[$legacyPlugin->$module_name->module->options_group_name]) : [];
```

Comment:
Maybe one option could be creating a recursive function that will run on that array checking each item, if it is an array, make a recursive call to the function. If not, apply the sanitize_text_field function to the value.

Reply:
I think i've gotten what you mean and updated the function, kindly check it out